guides/GUIDE_FOR_LIBRARY_AUTHORS.md

```diff
@@ -553,18 +555,39 @@ Controls whether the stack should be in `rtl` or `ltr` form.
 
 ### `disableBackButtonMenu` (iOS only)
 
-Boolean indicating whether to show the menu on longPress of iOS >= 14 back button.
+Boolean indicating whether to show the menu on longPress of the back button.
 
 ### `backButtonDisplayMode` (iOS only)
 
-Enum value indicating display mode of back button. It is used only when none of: `backTitleFontFamily`, `backTitleFontSize`, `disableBackButtonMenu` and `backTitleVisible=false` is set. The `backTitleVisible` forces `backButtonDisplayMode: minimal` and omits other values. Read more [#2800](https://github.com/software-mansion/react-native-screens/pull/2800). The other props, under the hood, customize `backButtonItem` which overrides `backButtonDisplayMode`. Read more [#2123](https://github.com/software-mansion/react-native-screens/pull/2123).
+Enum value indicating display mode of back button.
 
 Possible options:
 
-- `default` – show given back button previous controller title, system generic or just icon based on available space
-- `generic` – show given system generic or just icon based on available space
+- `default` – show given back button previous controller title, system generic or just icon based on available space and OS version
+- `generic` – show given system generic or just icon based on available space and OS version
 - `minimal` – show just an icon
 
+Starting from iOS 26:
+- the `title` of the previous screen is not used as back button title,
+- `generic` display mode behaves the same as `minimal`.
+
+In order to add text to the back button on iOS 26, use `default` display mode and specify text in `backTitle` property.
+
+On iOS versions prior to 26, using `generic` display mode with `backTitleFontFamily`, `backTitleFontSize`
+or `disableBackButtonMenu` property set is not supported due to limitations in the native platform.
+In such cases, display mode will fallback to `default`.
+
+If `backButtonUseModernImplementation` is set to `false`, this prop is used only when none of: `backTitleFontFamily`,
+`backTitleFontSize`, `disableBackButtonMenu` and `backTitleVisible=false` is set.
+
+### `backButtonUseModernImplementation` (iOS only)
+
+Used for backward compatibility only. Setting this prop to `false` is recommended.
```

Comment:
That's not recommended in every case - imo we should rather recommend migrating to the new implementation; we should be more specific about the versioning here. If we don't know which version of React Navigation it will be in, we should at least create a ticket to revisit it.